.github/workflows/deploy.yml

```diff
@@ -183,3 +183,56 @@ jobs:
       - name: Check that batches are being verified
         working-directory: ./scripts
         run: ./batch_verification_monitor.sh ${{ env.BATCH_VERIFICATION_MONITOR_TARGET }} ${{ env.BATCH_VERIFICATION_MONITOR_TIMEOUT }}
+
+  # Deploy the CDK environment against a local l1 chain with pre-deployed zkevm contracts.
+  pre-deployed-contracts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Install tools.
+      - name: Install kurtosis
+        run: |
+          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
+          sudo apt update
+          sudo apt install kurtosis-cli=${{ env.KURTOSIS_VERSION }}
+          kurtosis analytics disable
+
+      - name: Install yq
+        run: pip3 install yq
+
+      - name: Install foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      # Deploy components.
+      - name: Deploy L1 and zkEVM contracts
+        run: |
+          # Disable all deployment steps.
+          yq -Y --in-place 'with_entries(if .value | type == "boolean" then .value = false else . end)' params.yml
+          # Only deploy l1 and zkevm contracts.
+          yq -Y --in-place '(.deploy_l1 = true) | (.deploy_zkevm_contracts_on_l1 = true)' params.yml
```

Comment:
are these steps skipped if we're using pre-deployed contracts?

Reply:
Well in this PR I deploy the L1 with zkevm contracts in a first step and then, in a second step, I deploy the rest of the cdk/zkevm components by specifying the rollup manager address. I still need to test it against Sepolia and hope to add another job that skips the first two steps and targets Sepolia. I think that would be another great test case if that makes sense.